src/compiler/checker.ts

```diff
@@ -2194,10 +2194,17 @@ namespace ts {
                 }
             }
 
-            function buildDisplayForParametersAndDelimiters(parameters: Symbol[], writer: SymbolWriter, enclosingDeclaration?: Node, flags?: TypeFormatFlags, symbolStack?: Symbol[]) {
+            function buildDisplayForParametersAndDelimiters(thisType: Type, parameters: Symbol[], writer: SymbolWriter, enclosingDeclaration?: Node, flags?: TypeFormatFlags, symbolStack?: Symbol[]) {
                 writePunctuation(writer, SyntaxKind.OpenParenToken);
+                const useThisType = thisType && thisType.symbol;

```

Comment:
There are other types that don't have a symbol, e.g. union and intersection types. If you want to exclude the `void` type then you should check for that.